src/main/java/in/org/projecteka/hiu/consent/ConsentService.java

```diff
@@ -45,10 +48,13 @@
     }
 
     public Mono<Void> handleNotification(ConsentNotificationRequest consentNotificationRequest) {
-        // TODO: Need to figure out how we are going to figure out consent manager id.
-        // most probably need to have a mapping of @ncg = consent manager id
-        return validateRequest(consentNotificationRequest.getConsentRequestId())
-                .flatMap(consentRequest -> upsertConsentArtefacts(consentNotificationRequest).then());
+        if (consentNotificationRequest.getStatus() == ConsentStatus.GRANTED) {
+            // TODO: Need to figure out how we are going to figure out consent manager id.
+            // most probably need to have a mapping of @ncg = consent manager id
+            return validateRequest(consentNotificationRequest.getConsentRequestId())
+                    .flatMap(consentRequest -> upsertConsentArtefacts(consentNotificationRequest).then());
```

Comment:
This is applicable for all the notification right. I think we have to validate on all the scenarios.